formula-coroutines/src/main/java/com/instacart/formula/coroutines/SuspendAction.kt

```diff
@@ -0,0 +1,75 @@
+package com.instacart.formula.coroutines
+
+import com.instacart.formula.Action
+import com.instacart.formula.Cancelable
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ * Adapter which allows creating Formula [Action] from a Kotlin suspend function. Take a
+ * look at [SuspendAction.from] to create an [Action] from [suspend] type.
+ */
+interface SuspendAction<Event> : Action<Event> {
+
+    companion object {
+        /**
+         * Creates an [Action] which will launch a [suspend] created by factory function [create]
+         *
+         * ```
+         * SuspendAction.from { locationManager.currentLocation() }.onEvent { event ->
+         *   transition()
+         * }
+         * ```
+         */
+        fun <Event> from(
+            create: suspend () -> Event
+        ): SuspendAction<Event> {
```

Comment:
I think it should be `Result<Error>` to ensure that we emit errors.

Reply:
I think you mean `Result<Event>`, but shouldn't exception handling be up to the consumer? The consumer can handle wrapping their function in a try/catch and return a `Result` if they would like, but I'm not sure requiring that should be necessary. How do `FlowAction` and `RxAction` handle errors? It looks to me like unhandled errors in the stream would result in a crash.

Reply:
That's a good callout, I need to think about it more globally.

Reply:
It seems like having an api (`onError()`) on `ActionBuilder` might be a good idea.